indexing-service/src/main/java/io/druid/indexing/common/task/HadoopIndexTask.java

```diff
@@ -149,6 +158,13 @@ public String getClasspathPrefix()
   @Override
   public TaskStatus run(TaskToolbox toolbox) throws Exception
   {
+    //Setting up UsedSegmentLister which is used by DatasourcePathSpec to get the list of used
+    //segments
+    //Note that it is not possible to get UsedSegmentLister in the constructor as it is
+    //not available in overlord/mm but just peon. binding it in overlord/mm would require
+    //binding ToolboxFactory and all its dependencies which are really not needed in overlord/mm
+    DatasourcePathSpec.setUsedSegmentLister(globalInjector.getInstance(UsedSegmentLister.class));
```

Comment:
Would it work to change OverlordActionBasedUsedSegmentLister to take a TaskToolbox instead of a TaskToolboxFactory, and do: DatasourcePathSpec.setUsedSegmentLister(new OverlordActionBasedUsedSegmentLister(toolbox));

And avoid the global injector?